### PR TITLE
[feature][insert]Adapt the create table  statement to the nereids sql

### PR DESCRIPTION
**Pull request by wsjz**
## Proposed changes

issue: #31442

1. adapt  create table statement from doris  to hive
2. fix insert overwrite for table sink

> The doris create hive table statement:

```
mysql> CREATE TABLE buck2(
    ->     id int COMMENT 'col1',
    ->     name string COMMENT 'col2',
    ->     dt string COMMENT 'part1',
    ->     dtm string COMMENT 'part2'
    -> ) ENGINE=hive
    -> COMMENT "create tbl"
    -> PARTITION BY LIST (dt, dtm) ()
    -> DISTRIBUTED BY HASH (id) BUCKETS 16
    -> PROPERTIES(
    ->     "file_format" = "orc"
    -> );
```

> generated  hive create table statement:

```
CREATE TABLE `buck2`(
  `id` int COMMENT 'col1',
  `name` string COMMENT 'col2')
PARTITIONED BY (
 `dt` string,
 `dtm` string)
CLUSTERED BY (
  id)
INTO 16 BUCKETS
ROW FORMAT SERDE
  'org.apache.hadoop.hive.ql.io.orc.OrcSerde'
STORED AS INPUTFORMAT
  'org.apache.hadoop.hive.ql.io.orc.OrcInputFormat'
OUTPUTFORMAT
  'org.apache.hadoop.hive.ql.io.orc.OrcOutputFormat'
LOCATION
  'hdfs://HDFS8000871/usr/hive/warehouse/jz3.db/buck2'
TBLPROPERTIES (
  'transient_lastDdlTime'='1710840747',
  'doris.file_format'='orc')

```

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by wsjz** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/datasource/hive/HiveMetadataOps.java [hunk 1]
```diff
@@ -124,16 +134,45 @@ public void createTable(CreateTableStmt stmt) throws UserException {
...
+                    ddlProps.put(key, entry.getValue());
+                }
+            }
+            List<String> partitionColNames = new ArrayList<>();
+            if (stmt.getPartitionDesc() != null) {
+                partitionColNames.addAll(stmt.getPartitionDesc().getPartitionColNames());
+            }
+            HiveTableMetadata hiveTableMeta;
+            DistributionDesc bucketInfo = stmt.getDistributionDesc();
+            if (bucketInfo != null && Config.enable_external_hive_bucket) {
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/hive/HiveMetadataOps.java:154]: This if else is wrong, if Config.enable_external_hive_bucket is false and bucketInfo != null, should throw exception

File: fe/fe-common/src/main/java/org/apache/doris/common/Config.java [hunk 2]
```diff
@@ -2216,6 +2216,11 @@ public class Config extends ConfigBase {
             "Enable external table DDL"})
     public static boolean enable_external_ddl = false;
 
+    @ConfField(mutable = true, masterOnly = true, description = {
+            "启用Hive分桶表",
+            "Enable external hive bucket table"})
+    public static boolean enable_external_hive_bucket = false;
```
Comment (morningman) [inline fe/fe-common/src/main/java/org/apache/doris/common/Config.java:2222]:
```suggestion
    public static boolean enable_create_hive_bucket_table = false;
```

**Comment by morningman** [comment]
run buildall

**Comment by wsjz** [comment]
run buildall

**Comment by wsjz** [comment]
run buildall

**Comment by wsjz** [comment]
run buildall

**Comment by wsjz** [comment]
run buildall

**Comment by wsjz** [comment]
run buildall

**Comment by wsjz** [comment]
run buildall

**Comment by wsjz** [comment]
run p0

**Review by morningman — APPROVED** [review]

**Review by zddr — APPROVED** [review]

**Review by AshinGau — APPROVED** [review]

**Pull request merged**